### PR TITLE
Add freeton-solidity requirements and other utilities

### PR DESCRIPTION
**Pull request by hra687261**

### CODE REVIEW

**Comment by lefessan** [comment]
What is the `SetOfArgs` part used for ?

**Comment by hra687261** [comment]
It's used to hold the set of arguments of the "call" argument in the `buildExtMsg` primitive which looks something like: `call: {functionIdentifier [, list of function arguments]}`. Previously the arguments were parsed as an `ImmediateArray` which caused a typing error because the elements of the array are expected to be of the same type. Now they are parsed as a `SetOfArgs` which acts like a tuple. In fact I can replace it with a tuple but I wanted to distinguish between the two when they are printed.

**Comment by lefessan** [comment]
Are you using `ocp-indent` ? It looks like some of the changes are indentation only...

File: src/solidity-typechecker/solidity_tenv.ml [hunk 1]
```diff
@@ -62,7 +62,7 @@ let is_visible lookup visibility ~origin ~variable =
   | LStatic (Interface, _) ->
       !for_freeton
   | LStatic (Contract, false) ->
```
Comment (lefessan) [inline src/solidity-typechecker/solidity_tenv.ml:64]: Are you sure this is always true, or should be conditionned with `for_freeton` ?

Comment (hra687261) [reply src/solidity-typechecker/solidity_tenv.ml:64]: This one: `is_statically_visible visibility ~library:false`?
That's a mistake, as far as I know it should be `~library:true` even for freeton. I will change it.

**Comment by hra687261** [comment]
Yes I use `ocp-indent`. I think some files are not well indented but I will set them back as they were for now.

**Pull request merged**